bcl/src/bc_esp8266.c

```diff
@@ -667,6 +703,80 @@ static void _bc_esp8266_task(void *param)
 
                 continue;
             }
+            case BC_ESP8266_STATE_AP_AVAILABILITY_OPT_RESPONSE:
+            {
+                if (!_bc_esp8266_read_response(self) || memcmp(self->_response, "OK", 2) != 0)
+                {
+                    bc_esp8266_disconnect(self);
+                    return;
+                }
+
+                self->_state = BC_ESP8266_STATE_AP_AVAILABILITY_COMMAND;
+
+                continue;
+            }
+            case BC_ESP8266_STATE_AP_AVAILABILITY_RESPONSE:
+            {
+                /*
+                Success response:
+                +CWLAP:("Internet_7E",-74)
+                +CWLAP:("WLAN1-R87LDH",-77)
+                OK
+                */
+
+                self->_timeout_cnt++;
+                if (self->_timeout_cnt > _BC_ESP8266_TIMEOUT_WIFI_CONNECT)
+                {
+                    bc_esp8266_disconnect(self);
+                    return;
+                }
+
+                if (!_bc_esp8266_read_response(self))
+                {
+                    bc_scheduler_plan_current_from_now(_BC_ESP8266_DELAY_WIFI_CONNECT);
+                    return;
+                }
+
+                if (strcmp(self->_response, "OK\r") == 0)
+                {
+                    if (self->_event_handler != NULL)
+                    {
+                        self->_event_handler(self, BC_ESP8266_EVENT_AP_AVAILABILITY_RESULT, self->_event_param);
+                    }
+
+                    bc_esp8266_disconnect(self);
+                    return;
+                }
+                else if (strcmp(self->_response, "ERROR\r") == 0)
+                {
+                    bc_esp8266_disconnect(self);
+                    return;
+                }
+                else
+                {
+                    char text[76];
+                    sprintf(text, "+CWLAP:(\"%s\",", self->_config.ssid);
+                    size_t text_len = strlen(text);
+                    char *p = strstr(self->_response, text);
+                    if (p != NULL)
+                    {
+                        char *p2 = strchr(p + text_len, ')');
+                        if (p2 != NULL)
+                        {
+                            uint8_t rssi_len = p2 - (p + text_len);
+                            self->_message_buffer[0] = '1';
```

Comment:
From context of the code, I was not able to figure out what is this guy doing here :)